### ISSUE TITLE
Are generics supported?

### ISSUE THREAD

**Issue by overpas** [rickclephas/KMP-NativeCoroutines#101]
Is the plugin supposed to generate corresponding methods for an interface with parameterized types like this?
```kotlin
interface Feature<out Model, in Msg> : CoroutineScope {

    @NativeCoroutinesState
    val model: StateFlow<Model>

    fun dispatch(msg: Msg)
}
```
Is there a way to make this work?

**Comment by rickclephas** [comment]
Hi! Thanks for raising this question. The short answer is "kind of" 😅.

First of all there turns out to be an issue with the variance, which I am going to take a look at.
Once that is fixed the following answer will apply to your exact code snippet as well.

But for now I'll assume the code is as follows:
```kotlin
// file: Feature.kt
interface Feature<Model, Msg> : CoroutineScope {

    @NativeCoroutinesState
    val model: StateFlow<Model>

    fun dispatch(msg: Msg)
}
```

Where the plugin will generate the following extension properties:
```kotlin
// file: FeatureNative.kt
public val <Model, Msg> Feature<Model, Msg>.modelFlow: NativeFlow<Model>
  get() = model.asNativeFlow(null)

@ObjCName(name = "model")
public val <Model, Msg> Feature<Model, Msg>.modelValue: Model
  get() = model.value
```

So yeah technically generics are supported. However it doesn't really work as expected due to the following ObjC interop limitations:
* Generics aren't supported on protocols.
Meaning the generics are replaced by their upper bounds (in this case `Any?`).
* Extensions aren't supported on protocols.
Meaning the extensions are exposed on a `Kt` object (in this case `FeatureNativeKt`).

So if you would like to access the value of `model` in Swift you would need to do the following:
```swift
let feature: Feature = getFeature()
let model: Any? = FeatureNativeKt.model(feature)
```

You could add an extension in Swift to make this less annoying:
```swift
extension Feature {
    var model: Any? { FeatureNativeKt.model(self) }
}

let feature: Feature = getFeature()
let model: Any? = feature.model
```

That would "solve" the extension limitation, but it won't be able to solve the generics one.
 
The only way to workaround that limitation would be to add extensions to class types where you know the generic types:
```kotlin
abstract class SomeFeature: Feature<String, Int>
```

```swift
extension SomeFeature {
    var model: String { FeatureNativeKt.model(self) as! String }
}

let feature: SomeFeature = getSomeFeature()
let model: String = feature.model
```

**Comment by rickclephas** [comment]
FYI the issue with the variance has been fixed in [1.0.0-ALPHA-7](https://github.com/rickclephas/KMP-NativeCoroutines/releases/tag/v1.0.0-ALPHA-7).
